src/log.rs

```diff
@@ -477,8 +508,13 @@ impl Log {
 		Ok(bytes)
 	}
 
+	// Read from queue log enacted to db successfully,
+	// we therefore can remove them from the shared overlay.
 	pub fn end_read(&self, cleared: Cleared, record_id: u64) {
+		// [Review] this can happen because this is record_id from
+		// db queue and db_queue can fail so some id can be skippend.
 		if record_id >= self.next_record_id.load(Ordering::Relaxed) {
+			// [Review] racy?
```

Comment:
I think this may only happen when replaying the log initially. This is used to initialise `next_record_id` from an existing log.  After that it increments normally in `begin_record`. Not the most elegant solution and perhaps could be better solved by pre-scanning the log.